src/frontend-samples/read-settings-sample/ExternalSettingsUI.tsx

```diff
@@ -0,0 +1,169 @@
+          value = response.setting || ''
+      }
+
+      this.setState({
+        settingResult: response,
+        settingValue: value
+      });
+    });
+  }
+
+  // Handler to get settings value into state, when you modify textarea element in the dialog
+  handleSettingsValueChange(event: any) {
+    const settingValue = event.target.value;
+    this.setState({ settingValue: settingValue });
+  }
+
+  // The showcase does not have permission to write data, it is expected to fail with 403 Forbidden. 
+  // However saveSetting method will work in your project with signed-in user, who has required permissions in the project.
+  saveSettings() {
+    this.setState({ saveInProgress: true });
+    AuthorizedFrontendRequestContext.create().then(async context => {
+      const response = await IModelApp.settings.saveSetting(context, this.state.settingValue, namespace, this.state.settingKey!, true, this.state.projectId, this.state.imodelId);
+      this.setState({
+        settingResult: response,
+        saveInProgress: false
+      });
+    });
+  }
+
+  // Helper method to show status get/write operations with external setting in the dialog
+  private showStatus() {
+    if (!this.state.settingResult || this.state.settingResult.status == SettingsStatus.Success) {
+      return (<div></div>);
+    }
+
+    return (
+      <div style={{ lineBreak: 'anywhere', overflowWrap: 'break-word', maxWidth: '30vw' }}>
+        <SmallText style={{ color: 'var(--foreground-alert)' }}>
+          {`${this.state.settingResult.status} ${this.state.settingResult.errorMessage}`}
+        </SmallText>
+      </div>
+    );
+  }
+
+  /** Components for rendering the sample's instructions and controls */
+  private getControls() {
+    return (
+      <>
+        <LabeledSelect label="Settings Name:" options={["Json_Data", "Arbitrary_Text", "CSV_Data"]} placeholder="Browse by name..." value={this.state.settingKey} onChange={this.handleSettingsChange} />
+        <LabeledTextarea rows={10} key="test" label="Settings Value:" placeholder="" className="uicore-full-width" value={this.state.settingValue} onChange={this.handleSettingsValueChange} />
+        {this.showStatus()}
+        <div style={{ height: "35px", display: "flex", alignItems: "center", justifyContent: "center" }}>
+          {this.state.saveInProgress ?
+            (<div ><Spinner size={SpinnerSize.Small} /> Saving...</div>) :
+            <Button onClick={this.saveSettings} disabled={!this.state.settingKey}>Save settings</Button>
+          }
+        </div>
+        <DisabledText>Note: save doesn't work in this read-only environment.</DisabledText><br />
+        <DisabledText>Forbidden error is expected.</DisabledText>
+      </>
+    );
+  }
+
+  /** The sample's render method */
+  public render() {
+    return (
+      <>
+        <ControlPane instructions="Demonstrating read and write of custom settings" controls={this.getControls()} iModelSelector={this.props.iModelSelector}></ControlPane>
```

Comment:
Instructions should be in the form of telling the user what to do, ex. "Choose a Setting Name below to read that setting from the ProductSettingsService".

Reply:
Changed instructions as suggested